Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,321 @@
+
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+public class UIManager : MonoBehaviour
+{
+    [SerializeField] private TextMeshProUGUI summonName1;
+    [SerializeField] private TextMeshProUGUI summonName2;
+    [SerializeField] private TextMeshProUGUI summonName3;
+    [SerializeField] private TextMeshProUGUI summonName4;
+    [SerializeField] private TextMeshProUGUI summonName5;
+
+    [SerializeField] private Image summonImage1;
+    [SerializeField] private Image summonImage2;
+    [SerializeField] private Image summonImage3;
+    [SerializeField] private Image summonImage4;
+    [SerializeField] private Image summonImage5;
+
+    [SerializeField] private Sprite ghostImage;
+    [SerializeField] private Sprite skeletonImage;
+    [SerializeField] private Sprite zombieImage;
```

Comment:
Put all Sprite objets into a serialized list of sprites, and create a private dictionary with a string as the key and sprite as the value, use the summons class names as keys to get the sprites

Reply:
Done